data/transactions/logic/eval.go

```diff
@@ -1755,6 +1758,37 @@ func (cx *evalContext) assetParamsEnumToValue(params *basics.AssetParams, field
 	return
 }
 
+func (cx *evalContext) appParamsEnumToValue(params *basics.AppParams, creator basics.Address, field uint64) (sv stackValue, err error) {
+	switch AppParamsField(field) {
+	case AppApprovalProgram:
+		sv.Bytes = params.ApprovalProgram[:]
+	case AppClearStateProgram:
+		sv.Bytes = params.ApprovalProgram[:]
```

Comment:
shouldn't copy? If I remember right all txn fields used to be copied when exposed to TEAL

Reply:
Also the clear state program is returning the approval program.

Reply:
Fixed the Approval/Clear thing. We don't copy because the rule is that TEAL is not allowed to modify byteslices in place. assetParamsEnumToValue, for example, does things this way.

(You can change the sv.Bytes in an opcode, you just can't modify, say, sv.Bytes[2])